plasTeX/__init__.py

```diff
@@ -914,27 +914,97 @@ def digest(self, tokens):
         if dopars:
             self.paragraphs()
 
-
 class NoCharSubEnvironment(Environment):
     """
     A subclass of Environment which prevents character substitution inside
     itself.
     """
-    def __init__(self, *args, **kwargs):
-        # Will hold the owner document charsubs to restore it at the end
-        self.charsubs = []
-        super(NoCharSubEnvironment, self).__init__(*args, **kwargs)
+
+    def normalize(self, charsubs=None):
+        """ Normalize, but don't allow character substitutions """
+        return Environment.normalize(self)
```

Comment:
I think it would be much clearer to write:
```suggestion
        return Environment.normalize(self, charsubs=None)
```
rather than relying on the fact this is the default value anyway.